### ISSUE TITLE
Add smoke tests verifying basic functionality without any 3rd party deps

### ISSUE THREAD

**Issue by vytas7** [falconry/falcon#1790]
Starting from 2.0, Falcon (proudly) has no hard dependencies except the Python standard library.
We need to add smoke tests verifying basic functionality such as WSGI, ASGI, media handling, WebSocket etc without any other packages installed in the environment.

**Comment by CaselIT** [comment]
Yes, asgiref seems to be better, since it also supports wsgi; https://github.com/django/asgiref/#wsgi-to-asgi-adapter

**Comment by marco-georgaklis** [comment]
Hi! Another developer and I are interested in taking this issue on. Would you be able to provide more information on the issue?

**Comment by vytas7** [comment]
Glad to hear @marco-georgaklis, go ahead!

As to more information, you could first of all check if this isn't already covered by the existing `tox` environments.
The main idea behind this is that Falcon has no dependencies on any other 3rd party Python package, so it would be good to run a subset of tests/smoke tests requiring as little stuff in the venv in question as practically possible. Probably we can accept `pytest` and its deps, and maybe a WSGI/ASGI server like `uvicorn`, but we'd need to take a look at the respective indirect deps.

**Comment by marco-georgaklis** [comment]
So as an example, for the `smoke-test` tox environment, it looks like the requirements in [requirements/bench](https://github.com/falconry/falcon/blob/master/requirements/bench) are installed. Would the goal be to limit the number of dependencies (indirect and direct) in this file?

**Comment by marco-georgaklis** [comment]
@vytas7 Sorry to bother you again but just wanted to bump this. We have experimented with taking dependencies (i.e. vmprof) out of [requirements/bench](https://github.com/falconry/falcon/blob/master/requirements/bench) and have seemed to experience no issues with running the associated smoke tests (ran `tox -e py38_smoke'). Is this the correct workflow?
